src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,216 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.ColonyManager;
+import net.minecraft.command.ICommandSender;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+/**
+ * Get the Colony ID and Citizen ID out of a command.
+ */
+
+public final class GetColonyAndCitizen
+{
+
+    private static final String ONLY_NUMBERS                = "Please only use numbers for the %s ID!";
+    private static final String TOO_MANY_ARGUMENTS          = "Too many arguments!";
+    private static final String UNKNOWN_ERROR               = "Unknown Error!";
+    private static final String NOT_FOUND                   = "%s not found!";
+    private static final String NO_COLONY                   = "You haven't got a colony!";
+    private static final int    ONLY_NUMBERS_CODE           = -1;
+    private static final int    TOO_MANY_ARGUMENTS_CODE     = -2;
+    private static final int    UNKNOWN_ERROR_CODE          = -3;
+    private static final int    NOT_FOUND_CODE              = -4;
+    private static final int    NO_COLONY_CODE              = -5;
+    private static final int    NO_ARGUMENT_LENGTH          = 0;
+    private static final int    SHORT_ARGUMENT_LENGTH       = 1;
+    private static final int    NORMAL_ARGUMENT_LENGTH      = 2;
+    private static final int    NAME_ARGUMENT_LENGTH        = 3;
+    private static final int    ID_AND_NAME_ARGUMENT_LENGTH = 4;
+    private static final int    TOO_MANY_ARGUMENTS_LENGTH   = 5;
+
+    private GetColonyAndCitizen()
+    {
+        throw new IllegalAccessError("Utility Class");
+    }
+
+    /**
+     * Getting the colony ID.
+     * @param sender    The sender of the command
+     * @param args      The given arguments
+     * @return Returns colony ID (Integer)
+     */
+    public static int getColonyId(@NotNull final ICommandSender sender, @NotNull final String... args)
+    {
+        int colonyId;
+        final UUID mayorID = sender.getCommandSenderEntity().getUniqueID();
+        if (args.length == NORMAL_ARGUMENT_LENGTH || args.length == ID_AND_NAME_ARGUMENT_LENGTH)
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[NO_ARGUMENT_LENGTH]);
```

Comment:
This constant makes no sense here? What it the argument at position 0? Not NO_ARGUMENT_LENGTH because it is an argument (not no argument) and it is not a length. Confusing, better make a second constant